src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -1436,7 +1436,7 @@ public String create(final String path, byte data[], List<ACL> acl,
         request.setData(data);
         request.setFlags(createMode.toFlag());
         request.setPath(serverPath);
-        if (acl != null && acl.size() == 0) {
+        if (acl == null || acl.isEmpty() || acl.contains(null)) {
```

Comment:
Throwing IllegalArgumentException is the correct way but it may break the existing behavior. Consider the case where application code is like 

           zk.create(callBackHandler)` // call to create api and callback handler is passed
           doSomething();
           - - - -  // some more stuff

At present, if client pass null ACL parameter, the error will be handled by callBackHandler and in application code doSomething() method will execute. 

But if we throw IllegalArgumentException , then doSomething() method will not execute. So it will change the existing behavior of application code. 

Please suggest if we really need to handle this case.